client/main.lua

```diff
@@ -64,19 +64,27 @@ Skillbar.Repeat = function(data)
     end)
 end
 
-CreateThread(function()
-    while true do
-        if Skillbar.Data.Active then
-            if IsControlJustPressed(0, 38) then
-                SendNUIMessage({
-                    action = "check",
-                    data = Skillbar.Data.Data,
-                })
+function BarLoop()
+    if not Skillbar.Data.Active then
+        Skillbar.Data.Active = true
+        CreateThread(function()
+            while true do
+                if Skillbar.Data.Active then
+                    if IsControlJustPressed(0, 38) then
+                        SendNUIMessage({
+                            action = "check",
+                            data = Skillbar.Data.Data,
+                         })
+                    end
+                else
+                    looped = false
+                    break
+                end
```

Comment:
```suggestion
            while Skillbar.Data.Active do
                if IsControlJustPressed(0, 38) then
                    SendNUIMessage({
                        action = "check",
                        data = Skillbar.Data.Data,
                    })
                end
```
Remove extra condition check and unused variable